src/components/ScreenWrapper/index.js

```diff
@@ -21,149 +21,155 @@ import toggleTestToolsModal from '@userActions/TestTool';
+
+        const panResponder = useRef(
+            PanResponder.create({
+                onStartShouldSetPanResponderCapture: (e, gestureState) => gestureState.numberActiveTouches === CONST.TEST_TOOL.NUMBER_OF_TAPS,
+                onPanResponderRelease: toggleTestToolsModal,
+            }),
+        ).current;
+
+        const keyboardDissmissPanResponder = useRef(
+            PanResponder.create({
+                onMoveShouldSetPanResponderCapture: (e, gestureState) => {
+                    const isHorizontalSwipe = Math.abs(gestureState.dx) > Math.abs(gestureState.dy);
+                    const shouldDismissKeyboard = shouldDismissKeyboardBeforeClose && isKeyboardShown && Browser.isMobile();
+
+                    return isHorizontalSwipe && shouldDismissKeyboard;
+                },
+                onPanResponderGrant: Keyboard.dismiss,
+            }),
+        ).current;
+
+        useEffect(() => {
+            const unsubscribeTransitionEnd = navigation.addListener('transitionEnd', (event) => {
+                // Prevent firing the prop callback when user is exiting the page.
+                if (lodashGet(event, 'data.closing')) {
+                    return;
                 }
 
-                // We always need the safe area padding bottom if we're showing the offline indicator since it is bottom-docked.
-                if (includeSafeAreaPaddingBottom || (isOffline && shouldShowOfflineIndicator)) {
-                    paddingStyle.paddingBottom = paddingBottom;
+                setDidScreenTransitionEnd(true);
+                onEntryTransitionEnd();
+            });
+
+            // We need to have this prop to remove keyboard before going away from the screen, to avoid previous screen look weird for a brief moment,
+            // also we need to have generic control in future - to prevent closing keyboard for some rare cases in which beforeRemove has limitations
+            // described here https://reactnavigation.org/docs/preventing-going-back/#limitations
+            const beforeRemoveSubscription = shouldDismissKeyboardBeforeClose
+                ? navigation.addListener('beforeRemove', () => {
+                      if (!isKeyboardShownRef.current) {
+                          return;
+                      }
+                      Keyboard.dismiss();
+                  })
+                : undefined;
+
+            return () => {
+                unsubscribeTransitionEnd();
+
+                if (beforeRemoveSubscription) {
+                    beforeRemoveSubscription();
                 }
-
-                return (
-                    <View
-                        style={[styles.flex1, {minHeight}]}
-                        // eslint-disable-next-line react/jsx-props-no-spreading
-                        {...(isDevelopment ? panResponder.panHandlers : {})}
-                        testID={testID}
-                    >
+            };
```

Comment:
This did not work as intended. It later caused https://github.com/Expensify/App/issues/30268